src/cloudwatch-appsignals-mcp-server/awslabs/cloudwatch_appsignals_mcp_server/batch_processing_utils.py

```diff
@@ -0,0 +1,230 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities for interactive batch processing of audit operations."""
+
+import json
+import uuid
+from datetime import datetime, timezone
+from loguru import logger
+from typing import Any, Dict, List, Optional
+
+
+# Global storage for batch sessions (in production, this would be in a database)
+_batch_sessions: Dict[str, Dict[str, Any]] = {}
+
+# Configuration for session cache limits
+MAX_BATCH_SESSIONS = 1
+
+
+def create_batch_session(
+    targets: List[Dict[str, Any]],
+    input_obj: Dict[str, Any],
+    region: str,
+    banner: str,
+    batch_size: int = 10,
+    auto_complete: Optional[bool] = None,
+) -> str:
+    """Create a new batch processing session.
+
+    Args:
+        targets: List of all targets to process
+        input_obj: Base input object for API calls
+        region: AWS region
+        banner: Banner text for display
+        batch_size: Number of targets per batch
+        auto_complete: If True, process all batches automatically. If False, use interactive mode.
+                      If None, auto-decide based on target count.
+
+    Returns:
+        Session ID for tracking the batch processing
+    """
+    session_id = str(uuid.uuid4())
```

Comment:
Ideally, MCP server should be stateless. And it looks like you're generating random session id as key for the cache for each call. Does cache really work?